WalletWasabi/WabiSabi/Client/CredentialDependencies/DependencyGraph.cs

```diff
@@ -0,0 +1,443 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Client.CredentialDependencies
+{
+	public record DependencyGraph
+	{
+		public const int K = ProtocolConstants.CredentialNumber;
+
+		public static ImmutableArray<CredentialType> CredentialTypes { get; } = ImmutableArray.Create<CredentialType>(CredentialType.Amount, CredentialType.Vsize);
```

Comment:
FTR: It is possible to iterate enumerations so, this is not really required but it is okay.

Reply:
so change type to `IEnumerable`? that sounds right to me, the only goal of this property is to make the looping through credential types a bit easier and eliminate that ugly `NumTypes` hack I used

Reply:
No no, it was just a comment. What I say is that enumerations can be iterated and it is not necessary to fill a list or array with them in order to iterate. But forget it.

Reply:
ah I get it now, in that case this is completely unnecessary and i will remove